core/src/main/java/ai/timefold/solver/core/impl/move/streams/dataset/uni/AbstractForEachDataStream.java

```diff
@@ -38,15 +40,13 @@ public final void collectActiveDataStreams(Set<AbstractDataStream<Solution_>> da
     public final void buildNode(DataNodeBuildHelper<Solution_> buildHelper) {
         TupleLifecycle<UniTuple<A>> tupleLifecycle = buildHelper.getAggregatedTupleLifecycle(childStreamList);
         var outputStoreSize = buildHelper.extractTupleStoreSize(this);
-        var node = getNode(tupleLifecycle, outputStoreSize);
```

Comment:
Accessing the child class in this context appears to be an anti-pattern. It seems more effective to move it to `ForEachIncludingUnassignedUniNode`.